### PR TITLE
feat(storage): add blocking api for fuse source

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

File: src/query/storages/fuse/src/io/read/block_reader.rs [hunk 1]
```diff
@@ -344,6 +344,39 @@ impl BlockReader {
...
+        for index in indices {
+            let column_meta = &part.columns_meta[&index];
+
+            let op = self.operator.clone();
+
+            let location = part.location.clone();
+            let offset = column_meta.offset;
+            let length = column_meta.length;
+
+            threads.push(std::thread::spawn(move || {
```
Comment (zhang2014) [inline src/query/storages/fuse/src/io/read/block_reader.rs:369]: std::thread -> common_base::base::Thread

**Comment by Xuanwo** [comment]
The layer's issue will be resolved by https://github.com/datafuselabs/databend/pull/8421

File: src/query/storages/fuse/src/io/read/block_reader.rs (hunk 1 above)
Comment (Xuanwo) [reply src/query/storages/fuse/src/io/read/block_reader.rs:369]: Maybe we can use https://github.com/rayon-rs/rayon?

Comment (zhang2014) [reply src/query/storages/fuse/src/io/read/block_reader.rs]: common_base::base::Thread is used for resource tracking.

**Review by Xuanwo — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**